samples/routing/flowcontrol/Core_6/Shared/FlowControl/FlowManager.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using NServiceBus.Routing;
+
+class FlowManager
+{
+    public long GetNextMarker(string transportAddress)
+    {
+        var f = data.GetOrAdd(transportAddress, s => new FlowData());
+        return f.GenerateMarker();
+    }
+
+    public void Acknowledge(string key, string endpoint, int endpointInstanceHash, long markerValue)
+    {
+        FlowData f;
+        if (data.TryGetValue(key, out f))
+        {
+            f.Acknowledge(endpointInstanceHash, markerValue);
+        }
+    }
+
+    #region GetLeastBusy
+    public int? GetLeastBusyInstanceHash(string endpoint, IEnumerable<EndpointInstance> allInstances)
```

Comment:
`endpoint` doesn't appear to be used. Also the method that calls this is passed in an instance of `allInstances` that it sanitizes before passing it in here. Would it be better to do that sanitization here and just have the caller pass-through? I'm just not certain why the filter is necessary.